src/net/direct.c

```diff
@@ -43,6 +42,8 @@ static void direct_conn_send_internal(direct_conn dc)
        tcp_write or tcp_output, this will need to be revised to avoid
        deadlock. */
     spin_lock(&dc->send_lock);
+    if (q)
+        list_insert_before(&dc->sendq_head, &q->l);
```

Comment:
I might be missing something obvious, but why isn't this an append?

Reply:
This does add to the list, as the element is being placed "before" the list head (pointed to by head->prev). I am in favor of revising the list interface though...it could be much clearer.

Reply:
That's my bad! I didn't look closely enough at what a list head actually represents.